RingPublishingTracking/src/main/java/com/ringpublishing/tracking/internal/decorator/UserSsoDataDecorator.kt

```diff
@@ -0,0 +1,51 @@
+/*
+ *  Created by Grzegorz Małopolski on 10/6/21, 11:28 AM
+ * Copyright © 2021 Ringier Axel Springer Tech. All rights reserved.
+ *
+ */
+
+package com.ringpublishing.tracking.internal.decorator
+
+import android.util.Base64
+import com.google.gson.Gson
+import com.ringpublishing.tracking.data.Event
+import com.ringpublishing.tracking.internal.delegate.ConfigurationManager
+import com.ringpublishing.tracking.internal.log.Logger
+import java.io.UnsupportedEncodingException
+
+internal class UserSsoDataDecorator(private val configurationManager: ConfigurationManager, private val gson: Gson) : BaseDecorator()
+{
+	private val userId get() = configurationManager.getUserData().userId
+	private val ssoName get() = configurationManager.getUserData().ssoName ?: ""
+
+	override fun decorate(event: Event)
+	{
+		if (userId.isNullOrEmpty()) return
+
+		val encodedUserData = encodeUserSsoData(userId, ssoName)
+		encodedUserData?.let { event.add(EventParam.USER_SSO_DATA, it) }
+	}
+
+	private fun encodeUserSsoData(userId: String?, ssoName: String): String?
```

Comment:
user id moze byc bez ?

Reply:
sso name wsumie tez bo jak jest null dajesz ""

Reply:
Zmieniłem by przypisać go do stałej wewnątrz metody, to wtedy można dać bez ?